src/System.Numerics.Experimental/Ieee754Helpers.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+
+namespace System.Numerics.Experimental
+{
+    internal static class Ieee754Helpers
+    {
+        public const ulong DoubleSignMask = 0x80000000_00000000;
+        public const int DoubleSignShift = 63;
+        public const long DoubleExponentMask = 0x7FF00000_00000000;
+        public const int DoubleExponentShift = 52;
+        public const ulong DoubleSignificandMask = 0x000FFFFF_FFFFFFFF;
+        public const int DoubleSignificandShift = 0;
+
+        public const uint SingleSignMask = 0x80000000;
+        public const int SingleSignShift = 31;
+        public const int SingleExponentMask = 0x7F800000;
+        public const int SingleExponentShift = 23;
+        public const uint SingleSignificandMask = 0x7FFFFF;
+        public const int SingleSignificandShift = 0;
+
+        public const ushort HalfSignMask = 0x8000;
+        public const int HalfSignShift = 15;
+        public const ushort HalfExponentMask = 0x7C00;
+        public const int HalfExponentShift = 10;
+        public const ushort HalfSignificandMask = 0x03FF;
+        public const int HalfSignificandShift = 0;
+
+        public static double CreateDouble(ulong value)
```

Comment:
@tannergooding, Any opinion on the names of these methods (`Create***(*** value)`, `To****(*** value)`)? They don't really match the names on `BitConverter` class, but I thought it would be okay to omit the type being converted from, because the type can be inferred from the size of the type. It's just an internal class after all (which is another way of 'I was lazy typing all these `BitConverter.***ToInt**Bits`' 😅 ), but I'm happy to change them if you have any objections on it.

Reply:
I think they are fine for now, especially if the helper is internal.

I still need to open a proposal for exposing `SingleToUInt32Bits`, etc....

Reply:
That'd be lovely, converting back and forth to signed/unsigned types were pretty annoying!